src/core/mapping/base_mapper.cc

```diff
@@ -610,7 +610,11 @@ void BaseMapper::map_task(const MapperContext ctx,
 
     if (req_indices.empty()) {
       // This is a mapping for futures
-      output.future_locations.push_back(get_target_memory(task.target_proc, mapping.policy.target));
+      StoreTarget target = mapping.policy.target;
+#ifdef LEGATE_NO_FUTURES_ON_FB
```

Comment:
Note that this change is not strictly necessary to work around the assertion failure we're seeing currently. The creation of Future buffers on fbmem in `store.cc` is what's triggering it. However, since we know there's a general problem with Legion's handling of Futures instances on fbmem, I removed this case as well.